rest/src/main.rs

```diff
@@ -1121,16 +1138,160 @@ fn proposals() -> Value {
+                })),
+                "proof_specs": proof_specs,
+                "upgrade_path": client_state.upgrade_path,
+                "allow_update_after_expiry": raw_client_state.allow_update_after_expiry,
+                "allow_update_after_misbehaviour": raw_client_state.allow_update_after_misbehaviour,
+            }
+        },
+        "proof": null,
+        "proof_height": {
+            "revision_number": "0",
+            "revision_height": "0"
+        }
+    })
+}
+
+#[get("/ibc/core/channel/v1/connections/<connection>/channels")]
+async fn ibc_connection_channels(connection: &str) -> Value {
+    let channels = app_client()
+        .query(|app| {
+            app.ibc.ctx.query_connection_channels(EofTerminatedString(
+                IbcConnectionId::from_str(connection).unwrap(),
+            ))
+        })
+        .await
+        .unwrap();
+
+    let json_channels: Vec<_> = channels
+        .iter()
+        .map(|channel| {
+            json!({
+                "state": match channel.state {
+                    0 => "STATE_UNINITIALIZED_UNSPECIFIED",
+                    1 => "STATE_INIT",
+                    2 => "STATE_TRYOPEN",
+                    3 => "STATE_OPEN",
+                    i32::MIN..=-1_i32 | 4_i32..=i32::MAX => "STATE_UNINITIALIZED_UNSPECIFIED"
+                },
+                "ordering": match channel.ordering {
+                    0 => "ORDER_NONE_UNSPECIFIED",
+                    1 => "ORDER_UNORDERED",
+                    2 => "ORDER_ORDERED",
+                    i32::MIN..=-1_i32 | 3_i32..=i32::MAX => "ORDER_NONE_UNSPECIFIED"
+                },
+                "counterparty": channel.counterparty,
+                "connection_hops": channel.connection_hops,
+                "version": channel.version,
+                "port_id": channel.port_id,
+                "channel_id": channel.channel_id,
+            })
+        })
+        .collect();
+
+    json!({
+        "channels": json_channels,
+        "proof_height": {
+            "revision_number": "0",
+            "revision_height": "0"
+        },
+    })
+}
+
 #[get("/ibc/core/connection/v1/connections/<connection>")]
 async fn ibc_connection(connection: &str) -> Value {
     let connection = app_client()
-        .query(|app| app.ibc.ctx.query_connection(EofTerminatedString(IbcConnectionId::from_str(connection).unwrap())))
+        .query(|app| {
+            app.ibc.ctx.query_connection(EofTerminatedString(
+                IbcConnectionId::from_str(connection).unwrap(),
+            ))
+        })
         .await
         .unwrap()
         .unwrap();
 
+    let raw_connection = RawConnectionEnd::from(connection);
+
     json!({
-        "connection": RawConnectionEnd::from(connection),
+        "connection": {
+            "client_id": raw_connection.client_id,
+            "versions": raw_connection.versions,
+            "state": match raw_connection.state {
+                0 => "STATE_UNINITIALIZED_UNSPECIFIED",
```

Comment:
there's an enum in ibc-proto called State, which is a representation of i32, but I haven't figured out how to convert i32 to this State as it doesn't implement `From<i32>`, therefore this, please let me know if there's a more simple way